src/Chapter3.hs

```diff
@@ -480,6 +502,23 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+newtype Castle = Castle String
+data Building = Church | Library
+newtype House = House Int
+
+data City = City
+              { castle :: Castle
+              , wall :: Bool
+              , building :: Building
+              , houses :: [House]
+              }
+
+buildCastle :: String -> City -> City
+buildCastle castleName city = city { castle = Castle castleName }
+
+buildHouse :: Int -> City -> City
+buildHouse people city = city { houses = houses city ++ [ House people ] }
```

Comment:
According to our task, `House` can keep only from 1 to 4 people. You can use a Haskell design pattern called "smart constructor". To avoid building houses with more than 4 people you need to restrict usage of the Constructor and allow to create `House`s only with your `mkHouse` function.
You can read a bit more on the pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#smart-constructor

Generally, in here, you can go even further and you could restrict that on type level, by using something else than `Int` 🙂